server.go

```diff
@@ -112,7 +108,7 @@ func (s *Server) monitorHandler() {
 }
 
 func (s *Server) Handle(ctx context.Context, conn net.Conn) {
-	c := NewClient(conn, s.manager)
+	c := NewClient(ctx, conn, cfg)
```

Comment:
I'd consider making `NewClient` a method on `Server`. This way we wouldn't have to pass `cfg` everytime.

Reply:
This requires yet another refactoring. Let's not include it in the current PR and keep the PR as contained as possible.